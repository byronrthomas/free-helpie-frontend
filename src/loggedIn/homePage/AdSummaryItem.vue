<template>
  <div class="ad-summary-container">
    <div class="row">
      <div class="col-xs-3">
        <a href="unknown">{{ ad.postedBy }}</a>
      </div>
      <div class="col-xs-3">{{ ad.title }} </div>
      <div class="col-xs-3">{{ formattedSkills }}</div>
      <div class="col-xs-3">{{ ad.location }}</div>
    </div>
    <div class="row">
      <div class="col-xs-3 col-xs-offset-6">
        <button class="btn btn-primary" @click="$emit('viewPost', ad.id)">View</button>
      </div>
      <div class="col-xs-3">
        <button class="btn btn-primary" @click="$emit('savePost', ad.id)">Save</button>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: ['ad'],
  computed: {
    formattedSkills () {
      return this.ad.skills.length === 1
        ? this.ad.skills[0] 
        : this.ad.skills.reduce( (acc, nxt) => acc + ", " + nxt, '')
    }
  }
}
</script>

<style>
</style>
