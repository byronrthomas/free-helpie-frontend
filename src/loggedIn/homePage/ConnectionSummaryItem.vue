<template>
  <div class="row connection-summary">
    <div class="col-xs-2">
      <button class="btn btn-primary" @click="explainAction">{{nextAction}}</button>
    </div>
    <div class="col-xs-3">
      {{ withName }}
    </div>
    <div class="col-xs-5">
      {{ postSubject }}
    </div>
    <div class="col-xs-2"> {{formattedDate }}
    </div>
  </div>
</template>

<script>
import {formatDateTime} from '../../formatUtils/formatDateTime'

export default {
  props: {
    withName: String,
    postSubject: String,
    inviteSent: Date,
    nextAction: String
  },
  computed: {
    formattedDate () {
      return formatDateTime(this.inviteSent)
    }
  },
  methods: {
    explainAction () {
      if (this.nextAction === 'Accept invite') {
        alert('If you clicked this button, the invite would be accepted, the' +
        ' other user would be able to see your contact details and you could see theirs.' +
        ' This connection would become an Active connection. However, this isn\'t implemented yet.')
      } else if (this.nextAction === 'Cancel invite') {
        alert('If you clicked this button, your invite to connect would be canceled, without the other user being able to see your contact details.' +
        ' This connection would be deleted. However, this isn\'t implemented yet.')
      } else if (this.nextAction === 'Cancel sharing') {
        alert('If you clicked this button, you would stop sharing your contact details with the other user (however they might have already seen yours). You would not be able to see their details any more.' +
        ' This connection would be deleted. However, this isn\'t implemented yet.')
      }
    }
  }
}
</script>

<style>
  .connection-summary {
    border-top: 1px;
    border-bottom: 0px;
    border-left: 0px;
    border-right: 0px;
    border-style: solid;
    border-color: lightgray;
    padding: 5px
  }
</style>
