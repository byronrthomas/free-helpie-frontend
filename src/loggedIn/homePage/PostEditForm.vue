<template>
  <div style="background-color: #fffafa">
    <div class="row">
      <div class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3">
        <h4 style="color:rgb(235, 113, 180); text-align: center">{{ createOrUpdate }} post</h4>
      </div>
    </div>
    <form-segment :header-text="text.formTitle">
      <alerting-slot :alert-text="text.title.alert" :alerting="titleMissing">
        <div class="form-group">
          <label for="title">{{text.title.label}}</label>
          <input type="text" id="title" v-model="post.title" class="form-control">
        </div>
      </alerting-slot>
      <alerting-slot :alert-text="text.description.alert" :alerting="descriptionMissing">
        <div class="form-group">
          <label>{{text.description.label}}</label><br>
          <textarea v-model="post.description" :placeholder="text.description.placeholder"
          rows="5"
          class="form-control"/>
        </div>      
      </alerting-slot>
    </form-segment>
    <form-segment header-text="Location">
      <alerting-slot :alert-text="text.locations.alert" :alerting="locationsMissing">
        <label>{{text.locations.label}}</label>
        <div class="form-group">
          <label class="checkbox-inline">
              <input
                      type="checkbox"
                      id="remoteLocation"
                      value="true"
                      v-model="post.remote">Remote (over the phone / internet)
          </label>
        </div>
        <div class="form-group">
          <label>At specific locations:</label>
          <multi-select-fixed-options v-model="post.locations" :possibleOptions="possibleLocations"/>
        </div>
      </alerting-slot>
    </form-segment>
    <form-segment header-text="Interests &amp; skills">
      <alerting-slot :alert-text="text.skillsAndInterests.alert" :alerting="skillsAndInterestsMissing">
        <div class="form-group">
          <label>{{text.skillsAndInterests.interestsLabel}}</label>
          <multi-select-fixed-options v-model="post.interests" :possibleOptions="possibleInterests"/>
        </div>
        <div class="form-group">
          <label>{{text.skillsAndInterests.skillDetailLabel}}</label>
          <multi-select-fixed-options v-model="post.skills" :possibleOptions="possibleSkills"/>
        </div>
      </alerting-slot>    
    </form-segment>
    <form-segment header-text="Time required">
      <alerting-slot :alert-text="text.timeAmount.alert" :alerting="regularTimeMissing">
        <label>{{text.timeAmount.label}}</label>
        <div class="form-group">
          <select 
            id="timeAmount" 
            v-model="post.timings.regularAmount.unit">
              <option disabled value=""></option>
              <option>1hr</option>
              <option>2hr</option>
              <option>3hr</option>
              <option>4hr</option>
              <option>1day</option>
              <option>2days</option>
              <option>3days</option>
          </select>
          <span>per</span>
          <select 
            id="timeFrequency" 
            v-model="post.timings.regularAmount.frequency">
              <option disabled value=""></option>
              <option>Week</option>
              <option>Month</option>
          </select>          
        </div>
      </alerting-slot>
      <alerting-slot :alert-text="text.timeslots.alert" :alerting="timeslotsMissing">
        <div class="form-group">
          <label class="checkbox">{{text.timeslots.label}}</label>
          <label class="checkbox-inline">
              <input
                      type="checkbox"
                      id="weekday"
                      value="Weekday"
                      v-model="post.timings.slots"> Weekday
          </label>
          <label  class="checkbox-inline">
              <input
                      type="checkbox"
                      id="weekend"
                      value="Weekend"
                      v-model="post.timings.slots"> Weekend
          </label>
          <label  class="checkbox-inline">
              <input
                      type="checkbox"
                      id="evening"
                      value="Evening"
                      v-model="post.timings.slots"> Evening
          </label>
          <label class="checkbox-inline">
              <input
                      type="checkbox"
                      id="daytime"
                      value="Daytime"
                      v-model="post.timings.slots"> Daytime
          </label>
        </div>             
      </alerting-slot>
    </form-segment>
    <div class="row">
      <div class="col-xs-12 col-sm-8 col-sm-offset-2">
        <div class="row">
          <div class="col-xs-10 col-xs-offset-1">
            <div :class="alertClass" role="alert" v-if="!formContentsValid">
              <p>There are still some missing items, described above, you need to fill these out before you can submit</p>
            </div>
            <p v-if="lastServerError">{{ lastServerError }}</p>    
            <button 
              class="btn btn-primary" 
              style="width:100%; text-align:center"
              id="submitButton"
              :disabled="!formContentsValid"
              @click="submitForm">
              {{ createOrUpdate }}
              </button>        
            <br><br>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import MultiselectFixedOptions from '../../sharedComponents/MultiSelectFixedOptions.vue'
import FormSegment from '../../sharedComponents/FormSegment.vue'
import SlotWithReactiveAlert from '../../sharedComponents/SlotWithReactiveAlert.vue'
import { isValidPostType } from './postTypes'
import { getDescriptionsForPostType } from './postFormDescriptions'

function clonePost (input) {
  const result = {...input}
  result.locations = [...input.locations]
  result.skills = [...input.skills]
  result.interests = [...input.interests]
  return result
}

export default {
  props: {
    'initialPostData': {
      validator (value) {
        return Array.isArray(value.skills) &&
          Array.isArray(value.interests) &&
          Array.isArray(value.locations) &&
          (typeof value.title === 'string') &&
          (typeof value.description === 'string') &&
          (typeof value.remote === 'boolean') &&
          value.postedBy &&
          value.timings &&
          value.timings.regularAmount &&
          (typeof value.timings.regularAmount.unit === 'string') &&
          (typeof value.timings.regularAmount.frequency === 'string') &&
          Array.isArray(value.timings.slots) &&
          isValidPostType(value.postType)
      },
      required: true
    },
    'createOrUpdate': {
      type: String,
      validator (value) {
        return ['Create', 'Update'].includes(value)
      },
      required: true
    },
    'lastServerError': {
      type: String,
      default: ''
    },
    'possibleSkills': {
      type: Array,
      required: true
    },
    'possibleLocations': {
      type: Array,
      required: true
    },
    'possibleInterests': {
      type: Array,
      required: true
    }
  },
  data () {
    return {
      alertClass: {
        'alert': true,
        'alert-danger': true,
        'strong': true
      },
      text: getDescriptionsForPostType(this.initialPostData.postType),
      post: clonePost(this.initialPostData)
    }
  },
  computed: {
    skillsAndInterestsMissing () {
      return this.post.skills.length === 0 && this.post.interests.length === 0
    },
    locationsMissing () {
      return this.post.locations.length === 0 && !this.post.remote
    },
    regularTimeMissing () {
      return !this.post.timings.regularAmount.unit || !this.post.timings.regularAmount.frequency
    },
    timeslotsMissing () {
      return this.post.timings.slots.length === 0
    },
    titleMissing () {
      return !this.post.title
    },
    descriptionMissing () {
      return !this.post.description
    },
    formContentsValid () {
      return !(this.skillsAndInterestsMissing ||
        this.locationsMissing ||
        this.regularTimeMissing ||
        this.timeslotsMissing ||
        this.titleMissing ||
        this.descriptionMissing)
    },
    buttonText () {
      return this.createOrUpdate === 'Create' ? 'Post' : 'Save changes'
    }
  },
  components: {
    'multi-select-fixed-options': MultiselectFixedOptions,
    'form-segment': FormSegment,
    'alerting-slot': SlotWithReactiveAlert
  },
  methods: {
    submitForm () {
      console.log(this.post)
      this.$emit('submitForm', this.post)
    }
  }
}
</script>

<style>
</style>
