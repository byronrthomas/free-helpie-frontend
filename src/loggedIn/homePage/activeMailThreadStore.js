export function activeMailThreadStore (server) {
  return {
    namespaced: true,
    state: {
      activeThreads: []
    },
    getters: {
      activeThreads (state) {
        return state.activeThreads
      }
    },
    mutations: {
      setActiveThreads (state, mailitems) {
        state.activeThreads = mailitems
      }
    },
    actions: {
      refreshThreads ({commit, rootGetters, state}) {
        commit('setLastServerError', '', {root: true})
        server.get('/activeMailThreads', {authToken: rootGetters.authToken})
          .then(resp => commit('setActiveThreads', resp.data))
          .catch(err => commit('setLastServerError', err.message, { root: true }))
      }
    }
  }
}
