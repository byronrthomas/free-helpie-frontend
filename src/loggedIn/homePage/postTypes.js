export const HELP_OFFERED = 'helpOffered'
export const HELP_WANTED = 'helpWanted'
const VALID_TYPES = [HELP_OFFERED, HELP_WANTED]

export function isValidPostType (postType) {
  return VALID_TYPES.includes(postType)
}

