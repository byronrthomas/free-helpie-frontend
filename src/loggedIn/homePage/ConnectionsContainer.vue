<template>
  <div class="panel panel-primary">
    <div class="panel-heading">
      <h3 class="panel-title">{{ headerText }}</h3>
    </div>
    <div class="container table">
        <div class="row connections-header">
          <div class="col-xs-3 col-xs-offset-2"><strong>With</strong></div>
          <div class="col-xs-5"><strong>About post</strong></div>
          <div class="col-xs-2"><strong>Invite sent</strong></div>
        </div>
        <connection-summary-item
          v-for="(summary, index) in summaries"
          :key="index"
          :invite-sent="getActivityDate(summary)"
          :post-subject="getSubject(summary)"
          :next-action="nextAction"
          :with-name="getConversationName(summary)">
        </connection-summary-item>
      
    </div>
  </div>


</template>

<script>
import ConnectionSummaryItem from './ConnectionSummaryItem.vue'

export default {
  props: {
    summaries: Array,
    headerText: String,
    nextAction: String
  },
  methods: {
    getActivityDate (summary) {
      return summary.inviteSent
    },
    getConversationName (summary) {
      return summary.withName
    },
    getSubject (summary) {
      return summary.postSubject
    }
  },
  components: {
    'connection-summary-item': ConnectionSummaryItem
  }
}
</script>

<style>
  .connections-header {
    border-bottom: 1px;
    border-color: darkblue;
    border-left: 0px;
    border-right: 0px;
    border-style: solid;
    padding: 5px
  }
</style>
