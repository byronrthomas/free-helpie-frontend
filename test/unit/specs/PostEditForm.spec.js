import { mount } from 'vue-test-utils'
import PostEditForm from '@/loggedIn/homePage/PostEditForm.vue'
import { HELP_WANTED } from '../../../src/loggedIn/homePage/postTypes';

const CHECKED_DATA_ITEMS = [
  {
    propName: 'skillsAndInterestsMissing',
    fixes: [{
      name: 'some interest is present',
      data: {
        interests: ['anInterest']}},
    {
      name: 'some skill is present',
      data: {
        skills: ['aSkill']}}]},
  {
    propName: 'locationsMissing',
    fixes: [{
      name: 'remote is true',
      data: {
        remote: true}},
    {
      name: 'some location is present',
      data: {
        locations: ['aLocation']}}]},
  {
    propName: 'regularTimeMissing',
    fixes: [{
      name: 'both amount and frequency are non-empty',
      data: {
        timings: {
          regularAmount: {
            unit: 'Something',
            frequency: 'per something'},
          slots: []}}}]},
  {
    propName: 'timeslotsMissing',
    fixes: [{
      name: 'some timeslots are selected',
      data: {
        timings: {
          slots: ['Weekdays'],
          regularAmount: {
            unit: '',
            frequency: ''}}}}]},
  {
    propName: 'titleMissing',
    fixes: [{
      name: 'title is non-empty',
      data: {
        title: 'Help me please'}}]},
  {
    propName: 'descriptionMissing',
    fixes: [{
      name: 'description is non-empty',
      data: {
        description: 'Help me please'}}]}]

const EMPTY_POST =
  {
    postedBy: 'John Doe',
    title: '',
    interests: [],
    skills: [],
    locations: [],
    remote: false,
    description: '',
    timings: {regularAmount: {unit: '', frequency: ''}, slots: []},
    postType: HELP_WANTED
  }

const VALID_POST =
  {
    postedBy: 'John Doe',
    title: 'Help',
    interests: ['Interest'],
    skills: ['Skill'],
    locations: ['Location'],
    remote: false,
    description: 'Help me',
    timings: {regularAmount: {unit: '1h', frequency: 'Week'}, slots: ['Evenings']},
    postType: HELP_WANTED
  }
function makeMountArg (post) {
  const props = {
    initialPostData: post,
    createOrUpdate: 'Create',
    possibleSkills: [],
    possibleInterests: [],
    possibleLocations: []
  }
  return {propsData: props}
}

const getComponent = post => mount(PostEditForm, makeMountArg(post)).vm

describe('post edit form', () => {
  for (const check of CHECKED_DATA_ITEMS) {
    describe(`should report ${check.propName} correctly`, () => {
      it('should be true when they are missing', () => {
        const onTest = getComponent(EMPTY_POST)
        expect(onTest[check.propName]).toBeTruthy()
      })

      for (const fix of check.fixes) {
        it(`should be false when ${fix.name}`, () => {
          const fixedPost = {...EMPTY_POST, ...fix.data}
          const onTest = getComponent(fixedPost)
          expect(onTest[check.propName]).toBeFalsy()
        })
      }
    })
  }

  it(`should report formContentsValid as false whenever any single check is true`, () => {
    let post = EMPTY_POST
    for (const check of CHECKED_DATA_ITEMS.slice(0, CHECKED_DATA_ITEMS.length - 1)) {
      console.log('Fixing check ' + check.propName)
      post = {...post, ...check.fixes[0].data}
      const onTest = getComponent(post)
      expect(onTest.formContentsValid).toBeFalsy()
    }
  })

  it(`should report formContentsValid as true when all single checks are false`, () => {
    const onTest = getComponent(VALID_POST)
    for (const check of CHECKED_DATA_ITEMS) {
      if (onTest[check.propName]) {
        throw new Error(`Precondition failed, property ${check.propName} is true, should all be false`)
      }
    }
    expect(onTest.formContentsValid).toBeTruthy()
  })

  it('has the expected HTML when all checks are triggered (i.e. all data is missing)', () => {
    const onTest = mount(PostEditForm, makeMountArg(EMPTY_POST))
    for (const check of CHECKED_DATA_ITEMS) {
      if (!onTest.vm[check.propName]) {
        throw new Error(`Precondition failed, property ${check.propName} is false, should all be true`)
      }
    }
    expect(onTest.element).toMatchSnapshot()
  })

  it('has the expected HTML when all checks are inactive (i.e. all data is present)', () => {
    const onTest = mount(PostEditForm, makeMountArg(VALID_POST))
    if (!onTest.vm.formContentsValid) {
      throw new Error(`Precondition failed formContentsValid is false should be true`)
    }
    expect(onTest.element).toMatchSnapshot()
  })

  it('emits a postEditSubmitted event when the submit button is clicked', () => {
    const onTest = mount(PostEditForm, makeMountArg(VALID_POST))
    onTest.find('#submitButton').trigger('click')
    expect(onTest.emitted().submitForm).toBeTruthy()
    expect(onTest.emitted().submitForm.length).toBe(1)
    expect(onTest.emitted().submitForm[0]).toEqual([VALID_POST])
  })
})
