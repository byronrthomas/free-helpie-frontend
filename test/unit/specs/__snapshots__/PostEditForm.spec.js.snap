// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post edit form has the expected HTML when all checks are inactive (i.e. all data is present) 1`] = `
<div
  style="background-color: rgb(255, 250, 250);"
>
  <div
    class="row"
  >
    <div
      class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3"
    >
      <h4
        style="color: rgb(235, 113, 180); text-align: center;"
      >
        Create post
      </h4>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            What do you need
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <!---->
               
              <div
                class="form-group"
              >
                <label
                  for="title"
                >
                  Title
                </label>
                 
                <input
                  class="form-control"
                  id="title"
                  type="text"
                />
              </div>
               
              <!---->
               
              <div
                class="form-group"
              >
                <label>
                  Give some more details
                </label>
                <br />
                 
                <textarea
                  class="form-control"
                  placeholder="Give the details about what you need, things to think about:
- Telling us a bit about the background of why you need help
- Give plenty of detail about the activity
- Dig down into details of any specialist skills / experience needed
- Tell us what kind of person you would like to help out
- Be nice!"
                  rows="5"
                />
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Location
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <!---->
               
              <label>
                Where do you need help?
              </label>
               
              <div
                class="form-group"
              >
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="remoteLocation"
                    type="checkbox"
                    value="true"
                  />
                  Can help be given remotely?
      
                </label>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  At specific locations:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Interests & skills
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <!---->
               
              <div
                class="form-group"
              >
                <label>
                  People might want to help if they are interested in:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  Skills that your helpers will need:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Time required
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <!---->
               
              <label>
                How much time is required?
              </label>
               
              <div
                class="form-group"
              >
                <select
                  id="timeAmount"
                >
                  <option
                    disabled="disabled"
                    value=""
                  />
                   
                  <option>
                    1hr
                  </option>
                   
                  <option>
                    2hr
                  </option>
                   
                  <option>
                    3hr
                  </option>
                   
                  <option>
                    4hr
                  </option>
                   
                  <option>
                    1day
                  </option>
                   
                  <option>
                    2days
                  </option>
                   
                  <option>
                    3days
                  </option>
                </select>
                 
                <span>
                  per
                </span>
                 
                <select
                  id="timeFrequency"
                >
                  <option
                    disabled="disabled"
                    value=""
                  />
                   
                  <option>
                    Week
                  </option>
                   
                  <option>
                    Month
                  </option>
                </select>
              </div>
               
              <div
                class="form-group"
              >
                <!---->
                 
                <label
                  class="checkbox"
                >
                  What slots can people help with?
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="weekday"
                    type="checkbox"
                    value="TimeslotWeekday"
                  />
                   Weekday
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="weekend"
                    type="checkbox"
                    value="TimeslotWeekend"
                  />
                   Weekend
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="evening"
                    type="checkbox"
                    value="TimeslotEvening"
                  />
                   Evening
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="daytime"
                    type="checkbox"
                    value="TimeslotDaytime"
                  />
                   Daytime
      
                </label>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col-xs-12 col-sm-8 col-sm-offset-2"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-10 col-xs-offset-1"
        >
          <!---->
           
          <!---->
           
          <button
            class="btn btn-primary"
            style="width: 100%; text-align: center;"
          >
            
            Create
            
          </button>
           
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`post edit form has the expected HTML when all checks are triggered (i.e. all data is missing) 1`] = `
<div
  style="background-color: rgb(255, 250, 250);"
>
  <div
    class="row"
  >
    <div
      class="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3"
    >
      <h4
        style="color: rgb(235, 113, 180); text-align: center;"
      >
        Create post
      </h4>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            What do you need
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <div
                class="alert alert-danger strong"
                role="alert"
              >
                <p>
                  Give your post a title that will help them understand the outline of what you need
                </p>
              </div>
               
              <div
                class="form-group"
              >
                <label
                  for="title"
                >
                  Title
                </label>
                 
                <input
                  class="form-control"
                  id="title"
                  type="text"
                />
              </div>
               
              <div
                class="alert alert-danger strong"
                role="alert"
              >
                <p>
                  Be descriptive about the details of what you need
                </p>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  Give some more details
                </label>
                <br />
                 
                <textarea
                  class="form-control"
                  placeholder="Give the details about what you need, things to think about:
- Telling us a bit about the background of why you need help
- Give plenty of detail about the activity
- Dig down into details of any specialist skills / experience needed
- Tell us what kind of person you would like to help out
- Be nice!"
                  rows="5"
                />
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Location
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <div
                class="alert alert-danger strong"
                role="alert"
              >
                <p>
                  Please let your helpers know where they need to be - either add specific locations,
        or select remote, or both
      
                </p>
              </div>
               
              <label>
                Where do you need help?
              </label>
               
              <div
                class="form-group"
              >
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="remoteLocation"
                    type="checkbox"
                    value="true"
                  />
                  Can help be given remotely?
      
                </label>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  At specific locations:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Interests & skills
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <div
                class="alert alert-danger strong"
                role="alert"
              >
                <p>
                  Please add some interests or skills - the more you add, the easier it will be hard for helpers to match you
                </p>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  People might want to help if they are interested in:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
               
              <div
                class="form-group"
              >
                <label>
                  Skills that your helpers will need:
                </label>
                 
                <div>
                   
                  <select
                    class="form-control list-item"
                  >
                    <option
                      disabled="disabled"
                      value=""
                    />
                     
                  </select>
                   
                  <button
                    class="btn btn-primary"
                    disabled="disabled"
                  >
                    Add another
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div>
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-8 col-sm-offset-2"
      >
        <div
          class="floatable-bordered"
        >
          <h2
            style="text-align: center;"
          >
            Time required
          </h2>
           
          <br />
        </div>
         
        <div
          class="floatable-bordered floatable-bordered-bottom"
        >
          <br />
           
          <div
            class="row"
          >
            <div
              class="col-xs-10 col-xs-offset-1"
            >
              <div
                class="alert alert-danger strong"
                role="alert"
              >
                <p>
                  Please fill this out - it's important for your helpers to be able to match with you
        and plan their time
                </p>
              </div>
               
              <label>
                How much time is required?
              </label>
               
              <div
                class="form-group"
              >
                <select
                  id="timeAmount"
                >
                  <option
                    disabled="disabled"
                    value=""
                  />
                   
                  <option>
                    1hr
                  </option>
                   
                  <option>
                    2hr
                  </option>
                   
                  <option>
                    3hr
                  </option>
                   
                  <option>
                    4hr
                  </option>
                   
                  <option>
                    1day
                  </option>
                   
                  <option>
                    2days
                  </option>
                   
                  <option>
                    3days
                  </option>
                </select>
                 
                <span>
                  per
                </span>
                 
                <select
                  id="timeFrequency"
                >
                  <option
                    disabled="disabled"
                    value=""
                  />
                   
                  <option>
                    Week
                  </option>
                   
                  <option>
                    Month
                  </option>
                </select>
              </div>
               
              <div
                class="form-group"
              >
                <div
                  class="alert alert-danger strong"
                  role="alert"
                >
                  <p>
                    Please fill this out - helpers need to know what slots you need them
                  </p>
                </div>
                 
                <label
                  class="checkbox"
                >
                  What slots can people help with?
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="weekday"
                    type="checkbox"
                    value="TimeslotWeekday"
                  />
                   Weekday
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="weekend"
                    type="checkbox"
                    value="TimeslotWeekend"
                  />
                   Weekend
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="evening"
                    type="checkbox"
                    value="TimeslotEvening"
                  />
                   Evening
      
                </label>
                 
                <label
                  class="checkbox-inline"
                >
                  <input
                    id="daytime"
                    type="checkbox"
                    value="TimeslotDaytime"
                  />
                   Daytime
      
                </label>
              </div>
            </div>
          </div>
        </div>
         
        <br />
      </div>
    </div>
  </div>
   
  <div
    class="row"
  >
    <div
      class="col-xs-12 col-sm-8 col-sm-offset-2"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-10 col-xs-offset-1"
        >
          <div
            class="alert alert-danger strong"
            role="alert"
          >
            <p>
              There are still some missing items, described above, you need to fill these out before you can submit
            </p>
          </div>
           
          <!---->
           
          <button
            class="btn btn-primary"
            disabled="disabled"
            style="width: 100%; text-align: center;"
          >
            
            Create
            
          </button>
           
          <br />
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
